cart/views.py

```diff
@@ -1,3 +1,104 @@
-from django.shortcuts import render
+import json
+from json.decoder           import JSONDecodeError
 
-# Create your views here.
+from django.views           import View
+from django.http            import JsonResponse
+from django.core.exceptions import MultipleObjectsReturned
+
+from cart.models            import Cart
+from users.models           import User
+from products.models        import Product
+from core.utils             import login_decorator
+
+class CartView(View) :
+    @login_decorator
+    def post(self, request) :
+        try :
+            data       = json.loads(request.body)
+            user       = request.user
+            product_id = data['product_id']
+            quantity   = data['quantity']
+
+            if not Product.objects.filter(id=product_id).exists():
+                return JsonResponse({'message':'PRODUCT_NOT_EXIST'}, status=400)
+
+            if quantity <= 0:
+                return JsonResponse({'message':'QUANTITY_ERROR'}, status=400)
+        
+            cart, is_created  = Cart.objects.get_or_create(
+                user_id       = user.id,
+                product_id    = product_id
+            )
+            cart.quantity += quantity
+            cart.save()
+            
+            if is_created == 0 :
+                return JsonResponse({'message': 'UPDATE_SUCCESS'}, status=200)
+            return JsonResponse({'message': 'CREATE_SUCCESS'}, status=201)
+            
+        except Cart.DoesNotExist :
+            return JSONDecodeError({'message':'INVAILD_CART'}, status=400)
+        except JSONDecodeError :
+            return JsonResponse({'message':'JSON_DECODE_ERROR'}, status=400)
+        except KeyError :
+            return JsonResponse({'message':'KEY_ERROR'}, status=400)
+
+    @login_decorator
+    def get(self, request):
+        user  = request.user.id
+        carts = Cart.objects.filter(user_id=user)
```

Comment:
```suggestion
        carts = Cart.objects.filter(user=request.user)
```
user면 user, id면 id 명확하게 사용해야 합니다. user라는 변수에 request.user의 id를 저장하고
이후에 user_id=user 라는 코드를 보면 오해가 있을 수 있습니다. 파이썬은 타입을 강제할 수 없기 때문에 더욱 네이밍에 신경써야 합니다.
또한 현재 코드에서 굳이 변수에 지정해서 활용할 이유가 없어보입니다.